qcodes/instrument_drivers/ZI/ZIHDAWG8.py

```diff
@@ -177,21 +199,38 @@ def upload_sequence_program(self, awg_number: int,
         Returns:
             0 is Compilation was successful with no warnings.
             2 if Compilation was successful but with warnings.
+
+        Raises:
+            CompilerError: If error occurs during compilation of the sequence
+            program, or if a warning is elevated to an error.
         """
         self.awg_module.set('awgModule/index', awg_number)
         self.awg_module.set('awgModule/compiler/sourcestring', sequence_program)
         while len(self.awg_module.get('awgModule/compiler/sourcestring')
-                      ['compiler']['sourcestring'][0]) > 0:
-            time.sleep(0.1)
+                  ['compiler']['sourcestring'][0]) > 0:
+            pass
```

Comment:
isn't it more efficient (for some reason, i don't know) so still sleep here because the instrument is busy with compilation and if we are polling it constantly and very frequently, it slows down the compilation progress?

Reply:
Yes, that could be. When I thought what is proper sleeping time, I could only come up with None. But I'll reintroduce it as a private property of, say, 10ms.

Reply:
awesome :)